xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/extension/extension.js

```diff
@@ -495,6 +495,12 @@ XWiki.ExtensionBehaviour = Class.create({
       parameters : formData,
       onCreate : function() {
         dependency.removeClassName('extension-item-unknown').addClassName('extension-item-loading');
+        // Remove the unknown icon.
+        dependency.removeChild(dependency.childNodes[1]);
+        // Add the load icon
+        const loadPath = '$xwiki.getSkinFile("icons/xwiki/spinner.gif")';
+        const loadIcon = '<img  src="'+ loadPath + '" alt=""/>';
+        dependency.update(loadIcon + dependency.innerHTML);
```

Comment:
I'd rather use https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceWith to replace the unknown icon with the loading icon.